community/py-lz4/APKBUILD

```diff
@@ -1,40 +1,53 @@
 # Contributor: Stuart Cardall <developer@it-offshore.co.uk>
 # Maintainer: Stuart Cardall <developer@it-offshore.co.uk>
 pkgname=py-lz4
-pkgver=0.7.0
+pkgver=0.8.2
 _pkgname=${pkgname#py-}
 pkgrel=0
 pkgdesc="LZ4 Bindings for Python"
 url="https://pypi.python.org/pypi/lz4"
 arch="all"
 license="BSD"
-depends="python"
-depends_dev=""
-makedepends="python-dev py-setuptools lz4-dev"
-install=""
-subpackages=""
-source="https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/$_pkgname/$_pkgname-$pkgver.tar.gz
-	01-system_lz4.patch
-	02-tests.patch
-	"
-
+makedepends="python2-dev python3-dev py-setuptools lz4-dev"
+subpackages="py3-${pkgname/py-/}:_py3 py2-${pkgname/py-/}:_py2 $pkgname-tests:_tests"
+source="https://files.pythonhosted.org/packages/source/${_pkgname:0:1}/$_pkgname/$_pkgname-$pkgver.tar.gz"
 builddir="$srcdir"/$_pkgname-$pkgver
+
 build() {
 	cd "$builddir"
-	python setup.py build || return 1
+	python2 setup.py build || return 1
+	python3 setup.py build || return 1
 }
 
 package() {
+	mkdir -p "$pkgdir"
+}
+
+_py() {
+	local python=$1
+	pkgdesc="$pkgdesc - $python"
+	depends=$2
+	arch="all"
+	install_if="$pkgname=$pkgver-r$pkgrel $python"
 	cd "$builddir"
-	python setup.py install --prefix=/usr --root="$pkgdir" || return 1
+	$python setup.py install --prefix=/usr --root="$subpkgdir"
+}
+
+_py2() {
+	_py python2
+	replaces="$pkgname"
+}
```

Comment:
Please follow [APKBUILD snippet on wiki](https://wiki.alpinelinux.org/wiki/APKBUILD_examples:Python). `_py python2` is after `replaces=`, as the _last statement_ of the function, for a good reason. If it’s not the last, as in your case, and not followed by `|| return 1`, then it may silently fail (abuild doesn’t run with `set -e` yet). If it’s the _last statement_, as on the wiki, then the `_py()`’s return value is implicitly returned from the `_py2()` function and so if `_py()` returns non-zero status, it halts the abuild.